public/app/plugins/panel/dashlist/DashList.tsx

```diff
@@ -181,12 +183,16 @@ export function DashList(props: PanelProps<Options>) {
                 </a>
                 {showFolderNames && dash.folderTitle && <div className={css.dashlistFolder}>{dash.folderTitle}</div>}
               </div>
-              <IconButton
-                tooltip={dash.isStarred ? `Unmark "${dash.title}" as favorite` : `Mark "${dash.title}" as favorite`}
-                name={dash.isStarred ? 'favorite' : 'star'}
-                iconType={dash.isStarred ? 'mono' : 'default'}
-                onClick={(e) => toggleDashboardStar(e, dash)}
-              />
+              {config.featureToggles.starsFromAPIServer ? (
+                <StarToolbarButton group="dashboard.grafana.app" kind="Dashboard" name={dash.uid ?? ''} />
+              ) : (
+                <IconButton
+                  tooltip={dash.isStarred ? `Unmark "${dash.title}" as favorite` : `Mark "${dash.title}" as favorite`}
+                  name={dash.isStarred ? 'favorite' : 'star'}
+                  iconType={dash.isStarred ? 'mono' : 'default'}
+                  onClick={(e) => toggleDashboardStar(e, dash)}
+                />
+              )}
```

Comment:
Wonder if we can move all this logic into one component and merge StarButton with StarToolbarButton?